scipy/stats/_binned_statistic.py

```diff
@@ -568,8 +568,9 @@ def binned_statistic_dd(sample, values, statistic='mean',
     elif statistic == 'median':
         result.fill(np.nan)
         for i in np.unique(binnumbers):
-            for vv in xrange(Vdim):
-                result[vv, i] = np.median(values[vv, binnumbers == i])
+            if binnumbers is not None:
```

Comment:
`binnumbers` is unconditionally initialized to an array ~35 lines before this, so I fail to see how can this condition ever not be true.